content/en/docs/collector/building/_index.md

```diff
@@ -2,10 +2,70 @@
 title: Building custom components
 description: Instructions on how to build your own collector components
 weight: 90
+cSpell:ignore: mdatagen ottl
 ---
 
 The OpenTelemetry Collector can not only be extended by existing components, but
 also by custom components, that you develop and build on your own. Here you will
 find instructions on how to build some of those components. For additional
 details take a look into the documents contained within the
 [opentelemetry-collector-contrib repository](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/README.md).
+
+There are several
+[classes](https://github.com/open-telemetry/opentelemetry-collector/blob/67d37183e6ac9b7180fefc6dc3a55f2a75c12fba/cmd/mdatagen/main.go#L192)
+of OpenTelemetry components, such as
+
+- `receivers`: Scrapers and Listeners which
+  [ingest data](https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver#readme),
+  typically used to convert data from an external source to OTLP
+- `exporters`: Ways to
+  [export data](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter#readme)
+  to non-OTLP formats, vendor-specific backends, or other Observability tools
+- `processors`: Ways to
+  [process data](https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor#readme)
+  in a pipeline
+- `connectors`:
+  [Connects](https://github.com/open-telemetry/opentelemetry-collector/tree/main/connector#readme)
+  the output (exporter) of one pipeline to the input (receiver) of another in a
+  single component
+- `extensions`: Ways to
+  [augment the collector runtime](https://github.com/open-telemetry/opentelemetry-collector/blob/main/extension/README.md),
+  such as providing
+  [health checks](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/README.md)
+- `cmd`: Commands for building and/or maintain the collector
+- `pkg`:
+  [Packages](https://github.com/search?q=org%3Aopen-telemetry+%22class%3A+pkg%22&type=code)
+  for adding functionality to the collector, such as support for
+  [golden tests](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/golden#readme)
+  or
+  [OTTL functions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl#readme)
+
+Most components are registered using
```

Comment:
This is only relevant if people are adding components to contrib. Perhaps this doc needs to be clearer on what's the target audience? Am I reading this because I'm developing a proprietary component to be used with my own distribution, used internally at my company? Or is this about developing an open-source component that is meant to be distributed as part of contrib? I would say that this doc is for the first case, and would expect people to check the contribution guidelines in contrib if they are looking for the second case.

Reply:
> They might end up in the contrib repository, but I argue that there are many use cases were folks want to have their "own" thing that might not even be OSS.

I'll let @svrnm confirm, but I believe the focus should be those many use cases where folks want to have their "own" thing. We can keep contrib off this doc, as it would make it a lot simpler, while still pointing people to the contrib-specific guidelines at that repo.

Reply:
I thought about this, and sorry if this is different to what we have discussed before, but I lean towards "making it a lot simpler", so keeping things out of the docs that are contrib-specific. If we think that they would add additional value, we can still add them in later as well.